components/automate-cli/cmd/chef-automate/status.go

```diff
@@ -234,7 +288,236 @@ func runStatusCmd(cmd *cobra.Command, args []string) error {
 	}
 }
 
+func runStatusFromBastion(flags *StatusCmdFlags, nodeOpUtils NodeOpUtils, remoteExe RemoteCmdExecutor, printStatusOutput func(map[string][]*CmdResult, string, *cli.Writer)) error {
+
+	infra, _, err := nodeOpUtils.getHaInfraDetails()
+	if err != nil {
+		return err
+	}
+
+	if !flags.automate && !flags.chefServer && !flags.opensearch && !flags.postgresql {
+		if len(flags.node) != 0 {
+			return status.Errorf(status.InvalidCommandArgsError, "Please provide service flag")
+		}
+
+		flags.automate = true
+		flags.chefServer = true
+
+		if !nodeOpUtils.isManagedServicesOn() {
+			flags.opensearch = true
+			flags.postgresql = true
+		}
+	}
+
+	statusCmdResults := make(chan StatusCmdResult, 4)
+
+	runStatusCmdForFrontEnd(infra, flags, statusCmdResults, remoteExe)
+
+	if nodeOpUtils.isManagedServicesOn() {
+		statusCmdResults <- StatusCmdResult{}
+		statusCmdResults <- StatusCmdResult{}
+		err = getValueFromChannel(statusCmdResults, printStatusOutput)
+		if err != nil {
+			return err
+		}
+		return handleManagedServiceErrorForStatusCmd(flags)
+	}
+
+	runStatusCmdForBackend(infra, flags, statusCmdResults, remoteExe)
+
+	return getValueFromChannel(statusCmdResults, printStatusOutput)
+}
+
+func runStatusCmdForFrontEnd(infra *AutomateHAInfraDetails, flags *StatusCmdFlags, statusCmdResults chan StatusCmdResult, remoteExe RemoteCmdExecutor) {
+
+	if flags.automate {
+		nodeStatus(flags, AUTOMATE, infra, statusCmdResults, remoteExe)
+	} else {
+		statusCmdResults <- StatusCmdResult{}
+	}
+
+	if flags.chefServer {
+		flags.automate = false
+		nodeStatus(flags, CHEF_SERVER, infra, statusCmdResults, remoteExe)
+	} else {
+		statusCmdResults <- StatusCmdResult{}
+	}
+}
+
+func runStatusCmdForBackend(infra *AutomateHAInfraDetails, flags *StatusCmdFlags, statusCmdResults chan StatusCmdResult, remoteExe RemoteCmdExecutor) {
+
+	if flags.postgresql {
+		flags.automate = false
+		flags.chefServer = false
+		nodeStatus(flags, POSTGRESQL, infra, statusCmdResults, remoteExe)
+	} else {
+		statusCmdResults <- StatusCmdResult{}
+	}
+
+	if flags.opensearch {
+		flags.automate = false
+		flags.chefServer = false
+		flags.postgresql = false
+		nodeStatus(flags, OPENSEARCH, infra, statusCmdResults, remoteExe)
+	} else {
+		statusCmdResults <- StatusCmdResult{}
+	}
+}
+
+func nodeStatus(flags *StatusCmdFlags, nodeType string, infra *AutomateHAInfraDetails, statusCmdResults chan StatusCmdResult, remoteExe RemoteCmdExecutor) {
+	go func(nodeType string, flags StatusCmdFlags, statusCmdResults chan<- StatusCmdResult) {
+		writer := cli.NewWriter(os.Stdout, os.Stderr, os.Stdin)
+		remoteExe.SetWriter(writer)
+		nodeMap := constructNodeMapForStatus(&flags, infra)
+		cmdResult, err := remoteExe.ExecuteWithNodeMap(nodeMap)
+		statusCmdResults <- StatusCmdResult{
+			cmdResult: cmdResult,
+			writer:    writer,
+			nodeType:  nodeType,
+			err:       err,
+		}
+	}(nodeType, *flags, statusCmdResults)
+}
```

Comment:
dont use goroutine here, change this to a normal function and where you are calling nodeStatus call it with "go"

Reply:
Done